pkg/execution/state/redis_state/queue.go

```diff
@@ -1784,6 +1800,109 @@ func (q *queue) ConfigLease(ctx context.Context, key string, duration time.Durat
 	}
 }
 
+func (q *queue) getShards(ctx context.Context) (map[string]*QueueShard, error) {
+	m, err := q.r.Do(ctx, q.r.B().Hgetall().Key(q.kg.Shards()).Build()).AsMap()
+	if rueidis.IsRedisNil(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error fetching shards: %w", err)
+	}
+	shards := map[string]*QueueShard{}
+	for k, v := range m {
+		shard := &QueueShard{}
+		if err := v.DecodeJSON(shard); err != nil {
+			return nil, fmt.Errorf("error decoding shards: %w", err)
```

Comment:
should this quit all together if it fails to decode one shard?
or should it just move on, but log the error?

Reply:
This indicates a pretty serious issue so I'm thinking we should quit overall.  We should never really find ourselves in a state where shards are malformed, and if we do find ourselves there I doubt the integrity of the data overall.  WDYT?

Reply:
SGTM